Source/JavaScriptCore/wasm/WasmBBQJIT.cpp

```diff
@@ -4297,6 +4297,12 @@ class BBQJIT {
 
         ScratchScope<1, 0> scratches(*this, lhsLocation, rhsLocation, resultLocation);
         if (isSigned && !IsMod && !checkedForNegativeOne) {
+            // The following code freely clobbers wasmScratchGPR. This would be a bug if either of our operands were
+            // stored in wasmScratchGPR, which is the case if one of our operands is a constant - but in that case,
+            // we should be able to rule out this check based on the value of that constant above.
+            ASSERT(lhsLocation.asGPR() != wasmScratchGPR);
+            ASSERT(rhsLocation.asGPR() != wasmScratchGPR);
+
```

Comment:
Let's also add `ASSERT(!rhs.isConst())` and `ASSERT(!lhs.isConst())`.

Reply:
Added in latest push.